dts/bindings/sensor/nxp,s32k-qdec.yaml

```diff
@@ -0,0 +1,112 @@
+            <1 TRGMUX_IP_OUTPUT_EMIOS0_CH5_9_IPP_IND_CH7 TRGMUX_IP_INPUT_LCU1_LC0_OUT_I3>,
+            <2 TRGMUX_IP_OUTPUT_LCU1_0_INP_I0            TRGMUX_IP_INPUT_SIUL2_IN2>,
+            <3 TRGMUX_IP_OUTPUT_LCU1_0_INP_I1            TRGMUX_IP_INPUT_SIUL2_IN3>;
+        trgmux-maps = < 54 3 >, /* TRGMUX_OUT_37 <-> eMIOS0_CH6 */
+           < 55 4 >; /* TRGMUX_OUT_38 <-> eMIOS0_CH7 */
+        lcu_inst = <1>;
+        lcu-output-filter-config =
+            /* LCU Out HW ID, Rise Filter, Fall Filter */
+            <0 5 5>, /* LCU O0 */
+            <1 5 5>, /* LCU O1 */
+            <2 2 2>, /* LCU O2 */
+            <3 2 2>; /* LCU O3 */
+        emios_inst = <0>;
+        emios-channels = <6 7>;
+   };
+
+compatible: "nxp,qdec-s32k"
+
+include: [pinctrl-device.yaml, sensor-device.yaml]
+
+properties:
+  micro-ticks-per-rev:
+    type: int
+    description: |
+      This is a number that is used to determine how many revolutions * 1000000
+      were done based on the current counter's value.
+
+  trgmux-inst:
+    type: int
+    enum: [0, 1]
+    description: |
+      This gives the hardware instance of TRGMUX module.
+
+  trgmux-io-config:
+    type: array
+    description: |
+      This gives the logic triggers configuration of TRGMUX module.
+      It contains 3 values for each of the 4 logic triggers used:
+        logic trigger number, output, input.
+      Hence, it's length should be '12'.
+      Ex:
+        trgmux-io-config =
+          <0 TRGMUX_IP_OUTPUT_EMIOS0_CH5_9_IPP_IND_CH6 TRGMUX_IP_INPUT_LCU1_LC0_OUT_I2>,
+          <1 TRGMUX_IP_OUTPUT_EMIOS0_CH5_9_IPP_IND_CH7 TRGMUX_IP_INPUT_LCU1_LC0_OUT_I3>,
+          <2 TRGMUX_IP_OUTPUT_LCU1_0_INP_I0            TRGMUX_IP_INPUT_SIUL2_IN2>,
+          <3 TRGMUX_IP_OUTPUT_LCU1_0_INP_I1            TRGMUX_IP_INPUT_SIUL2_IN3>;
+
+  lcu_inst:
+    type: int
+    description: |
+      This gives the hardware instance of LCU module.
+
+  emios_inst:
+    type: int
+    description: |
+      This gives the hardware instance of eMIOS module.
+
+  lcu-output-filter-config:
+    type: array
+    description: |
+      This array gives the configuration for each of the four outputs of LCU module.
+      It contains the following for each output: hardware output id, rise filter and fall filter.
+      The filters specify the delay in terms of CORE_CLK between the input and output line of LC.
+        We use this delay to generate short pulses at the rising and falling edges of input pulse.
+      It's length should be '12' - 3 entries for each of the four LCU outputs.
+      Ex: lcu-output-filter-config =
+        /* LCU Out HW ID, Rise Filter, Fall Filter */
+        <0 5 5>, /* LCU O0 */
+        <1 5 5>, /* LCU O1 */
+        <2 2 2>, /* LCU O2 */
+        <3 2 2>; /* LCU O3 */
+
+  emios-channels:
+    type: array
+    description: |
+      This is the array containing 2 emios channel TypeG numbers used by the qdec.
+      See Table 342 in S32K3XXRM for more information
+
+  trgmux-maps:
```

Comment:
If see this is used to configure pinmuxing of eMIOS or TRGMUX peripheral: https://github.com/zephyrproject-rtos/zephyr/blob/a53e882ff3777a3c67c03da984106a4625f56ebb/drivers/sensor/qdec_s32k/qdec_s32k.c#L151
There's already a pinctrl driver for that purpose, please add the needed pinmuxing configuration using `pinctrl` node in the board's overlay, and remove this property.

Reply:
I'll need to add new pinmux entries in - hal_nxp/dts/nxp/s32/S32/S32K344-172MQFP-pinctrl.h 
I hope that's ok ?

Reply:
fixed !

Reply:
fixed !

Reply:
The binding property and board configuration is still there, https://github.com/zephyrproject-rtos/zephyr/blob/2a924701eda579fb9c526edab42d1f37478c90cd/boards/arm/mr_canhubk3/mr_canhubk3.dts#L94-L96 pls double check

Reply:
Changed as per suggestions !